File: code-coaches/hard/security/Security.cs
using System;

class Security
{
    static bool IsMoneySafe(string layout)
    {
        // Find the indices of the money, thief, and guards on the layout
        int moneyIndex = layout.IndexOf("$");
        int thiefIndex = layout.IndexOf("T");
        int[] guardIndices = new int[layout.Length];
        int guardCount = 0;

        for (int i = 0; i < layout.Length; i++)
        {
            if (layout[i] == 'G')
                guardIndices[guardCount++] = i;
        }

        Array.Resize(ref guardIndices, guardCount);

        // Check if there is a guard between the money and the thief
        foreach (int guardIndex in guardIndices)
        {
            if (Math.Min(moneyIndex, thiefIndex) < guardIndex && guardIndex < Math.Max(moneyIndex, thiefIndex))
            {
                // Found a guard between the money and the thief
                return true;
            }
        }

        // No guard found between the money and the thief
        return false;
    }

    static void Main()
    {
        string layout = Console.ReadLine();

        if (IsMoneySafe(layout))
            Console.WriteLine("quiet");
        else
            Console.WriteLine("ALARM");
    }
}

File: code-coaches/hard/security/security.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool is_money_safe(char *layout)
{
    // Find the indices of the money, thief, and guards on the casino floor
    int money_index = strchr(layout, '$') - layout;
    int thief_index = strchr(layout, 'T') - layout;

    // Find all guard indices
    int guard_indices[strlen(layout)];
    int num_guards = 0;
    for (int i = 0; i < strlen(layout); i++)
    {
        if (layout[i] == 'G')
        {
            guard_indices[num_guards] = i;
            num_guards++;
        }
    }

    // Check if there is a guard between the money and the thief
    for (int i = 0; i < num_guards; i++)
    {
        int guard_index = guard_indices[i];
        if ((money_index < thief_index && money_index < guard_index && guard_index < thief_index)
            || (thief_index < money_index && thief_index < guard_index && guard_index < money_index))
        {
            return true;
        }
    }

    return false;
}

int main()
{
    char layout[1000];
    scanf("%s", layout);

    if (is_money_safe(layout))
        printf("quiet");
    else
        printf("ALARM");

    return 0;
}
